internal/data/data_manager_test.go

```diff
@@ -35,6 +36,47 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+func TestCacheInitFail(t *testing.T) {
+	cacheInitError := errors.New("Initialization error.")
+	coreconfig.Reset()
+	config.Set(coreconfig.MessageWriterCount, 1)
+	ctx := context.Background()
+	mdi := &databasemocks.Plugin{}
+	mdi.On("Capabilities").Return(&database.Capabilities{
+		Concurrency: true,
+	})
+	mdx := &dataexchangemocks.Plugin{}
+	ns := &core.Namespace{Name: "ns1", NetworkName: "ns1"}
+
+	vErrcmi := &cachemocks.Manager{}
+	vErrcmi.On("GetCache", cache.NewCacheConfig(
+		ctx,
+		coreconfig.CacheValidatorSize,
+		coreconfig.CacheValidatorTTL,
+		ns.Name,
+	)).Return(nil, cacheInitError)
+	_, err := NewDataManager(ctx, ns, mdi, mdx, vErrcmi)
+	assert.Equal(t, cacheInitError, err)
+	vErrcmi.AssertNumberOfCalls(t, "GetCache", 1)
```

Comment:
You can use the `.On().Return().Once()` syntax on the end of your expectations, coupled with `vErrcmi.AssertExpectations(t)` here to enforce that it was called exactly once.

Reply:
Thank you. Learnt something new today 🎉 I'll update the test cases.